compiler/src/dotty/tools/dotc/transform/SyntheticMembers.scala

```diff
@@ -232,35 +232,65 @@ class SyntheticMembers(thisPhase: DenotTransformer) {
     /** The class
      *
      *  ```
-     *  package p
-     *  case class C(x: T, y: T)
+     *  case object C
+     *  ```
+     *
+     *  gets the `hashCode` method:
+     *
+     *  ```
+     *  "C".hashCode // constant folded
+     *  ```
+     *
+     *  The class
+     *
+     *  ```
+     *  case class C(x: T, y: U)
+     *  ```
+     *
+     *  if non of `T` or `U` are primitive types, gets the `hashCode` method:
+     *
+     *  ```
+     *  def hashCode: Int = ScalaRunTime._hashCode(this)
+     *  ```
+     *
+     *  else if either `T` or `U` are primitive, gets the `hashCode` method implemented by [[caseHashCodeBody]]
+     */
+    def chooseHashcode(implicit ctx: Context) = {
+      if clazz.is(ModuleClass) then
```

Comment:
Please use the regular `if (...)` syntax to be consistent with the rest of the compiler, same with `for` below.

Reply:
fixed in https://github.com/lampepfl/dotty/pull/7019/commits/e9ce0332806b0189aa835c55e127a6f1f0795d9d